org.hl7.fhir.validation/src/main/java/org/hl7/fhir/validation/instance/utils/NodeStack.java

```diff
@@ -64,7 +64,12 @@ public NodeStack(IWorkerContext context, Element element, String refPath, String
     this.context = context;
     ids = new HashMap<>();
     this.element = element;
-    literalPath = refPath + "->" + element.getName();
+    int i = element.getName().indexOf(".");
+    if (i == -1) {
+      literalPath = refPath+".resolve().ofType(" + element.getName()+")";      
```

Comment:
Do elements have path formatted names?

Patient.name with refpath `Some.ref.path` becomes `Some.ref.path.resolve().ofType(Patient).name`?

Reply:
well, they do. It was -> instead of .resolve() which is illegal, so I'm just normalising that. I looked at reworking the way element paths happen completely but it was a harder problem than I expected, so I backed out those changes. maybe another day